examples/layers/rendering/src/binding.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/rendering.dart';
+
+/// An extension of [RenderingFlutterBinding] that owns and manages a
+/// [renderView].
+///
+/// Unlike [RenderingFlutterBinding], this binding also creates and owns a
+/// [renderView] to simplify bootstrapping for apps that have a dedicated main
+/// view.
+class ViewRenderingFlutterBinding extends RenderingFlutterBinding {
```

Comment:
From my understanding, this pull request results in breaking changes for `RenderingFlutterBinding`. `ViewRenderingFlutterBinding` "brings back" the old `RenderingFlutterBinding`.

Does anyone use `RenderingFlutterBinding` directly other than our example apps? Should we add `ViewRenderingFlutterBinding` to the framework so that folks using `RenderingFlutterBinding` can 
 use it? Or alternatively, should we make `RenderingFlutterBinding` maintain its existing behavior but give the framework some other API for a `RenderingFlutterBinding` variant that doesn't own a `renderView`?

Reply:
Your understanding is correct. I haven't found any usage for `RenderingFlutterBinding` outside of our examples; it is a bit of a rarity to use the rendering layer directly without widgets. My thinking was that we are removing the single `renderView` assumption from all other parts of the framework and it seemed awkward to leave it in the little-used `RenderingFlutterBinding`. Instead, I wanted to turn the `RenderingFlutterBinding` into something that can also be used properly in a multi-view environment. 

I initially considered adding `ViewRenderingFlutterBinding` to the framework, but ultimately (given the non-existent usage of `RenderingFlutterBinding`) I felt it wasn't worth the maintenance burden and instead just moved it out to the examples. If somebody really needs it, the can copy in from there...

Reply:
> but ultimately (given the non-existent usage of RenderingFlutterBinding) I felt it wasn't worth the maintenance burden

I agree on a technical level but I feel a bit uneasy on a philosophical level. We've advertised that Flutter is layered and that one can throw away the widgets layer if they'd like to build their own framework. Today, a "hello world" app that uses the rendering layer directly is remarkably succinct. Now, you'll need to do some non-trivial bootstrapping from `ViewRenderingFlutterBinding`.

Reply:
I would argue that the extra "boilerplate" is required because you need to decide if/how you want to support multiple renderViews. That is a decision the rendering layer explicitly leaves to a higher level abstraction build on top of it (see docs on `RendererBinding`). So, if you build on top of the rendering layer directly, you'd have to make that decision.

The `ViewRenderingFlutterBinding` (built on top of the rendering layer) now made the opinionated decision to own a specialized RenderView that is rendered into the implicit view. This works for some use cases, but not all use cases have a dedicated main render view. They may want to handle this differently.

The guiding philosophy for the rendering layer for me was that it does not own any RenderViews directly (and it doesn't care who or what created them). It only offers APIs to manage RenderViews that something else has created.

All philosophy aside, though, if we were to discover that building directly on top of the rendering layer while having a dedicated main view is a frequent use case I would totally go ahead and add the `ViewRenderingFlutterBinding` to the framework. But currently I don't think it is worth the effort...

Reply:
> The ViewRenderingFlutterBinding (built on top of the rendering layer) now made the opinionated decision to own a specialized RenderView that is rendered into the implicit view

That resonates with me: it's an opinionated decision that a "real" app/framework would not want to make once multi-view lands. Thanks for detailing your thoughts, I'll mark this as resolved.